platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -933,14 +919,14 @@ def get_revision(self):
         Returns:
             string: Revision value of device
         """
-        if self.dmi_data is None:
-            self.dmi_data = self._parse_dmi(DMI_FILE)
+        if self.vpd_data is None:
```

Comment:
if not self.vpd_data. This will check both None and empty dictionary.

Reply:
Fixed